pystac_client/client.py

```diff
@@ -148,20 +155,19 @@ def open(
             request_modifier=request_modifier,
             stac_io=stac_io,
         )
-        search_link = client.get_search_link()
-        # if there is a search link, but no conformsTo advertised, ignore
-        # conformance entirely
-        # NOTE: this behavior to be deprecated as implementations become conformant
-        if client._stac_io and (
-            ignore_conformance
-            or (
-                client
-                and "conformsTo" not in client.extra_fields.keys()
-                and search_link
-                and search_link.href
-                and len(search_link.href) > 0
-            )
+
+        if client._stac_io and ignore_conformance:
+            client._stac_io.set_conformance(None)
+        elif client._stac_io and (
+            "conformsTo" not in client.extra_fields.keys() and client._get_search_href()
         ):
+            warnings.warn(
+                (
+                    "Client does not have conformance set. This is equivalent to "
+                    "setting `ignore_conformace=True`"
```

Comment:
```suggestion
                    "Server does not advertise any conformance classes. This is equivalent to "
                    "setting `ignore_conformace=True` on the client."
```

It's really the server that doesn't advertise any conformance classes, right?

Reply:
Yeah that is right. I guess I meant the `Client` object but what you have makes more sense.